### ISSUE TITLE
Add CI verification step

### ISSUE THREAD

**Issue by pitkley** [greshake/i3status-rust#40]
PRs (and `master` for that matter) are currently not checked for anything, potentially causing [`master` not to build](https://github.com/greshake/i3status-rust/issues/30#issuecomment-304526359).

I propose to add a simple CI configuration, using one of the numerous free CI solutions like [Travis CI](https://travis-ci.org) or [Circle CI](https://circleci.com/).

-----

Following is a very simple Travis configuration (`.travis.yml`) which at least checks if a commit builds successfully:

```yaml
language: rust
rust:
  - stable
  - beta
  - nightly
os:
  - linux
matrix:
  allow_failures:
    - rust: nightly
script:
  - cargo build -v
  - cargo test
```

(I have added `cargo test` although there aren't any tests yet. This will take the burden of remembering to add a test-run to the CI configuration once tests might be added.)

**Comment by greshake** [comment]
I'll set it up when I have the time. Currently I check every PR by hand :)

**Comment by greshake** [comment]
Okay, I've got Travis running now. Do you know how to trigger builds for exisiting PR's right now? I'll just try to trigger it with a dummy commit to one of your PR's.

**Comment by pitkley** [comment]
Thanks for adding Travis. I see that you have removed `beta` from the Rust version matrix -- is there a specific reason why you did this? While compatibility with `nightly` probably includes `beta`, I would assume that a build failure on `beta` should not be allowed.

**Comment by greshake** [comment]
Yea Travis just didn't trigger because these PR's were created before I set it up. Happens automatically in the future. I removed beta because I assume that failure to build in beta either indicates an error in the beta toolchain (in which case I would have to spend unnessesary resources finding that error) or behaviour which will be deprecated in the future, in which case nightly build failure is a better indicator. Also it might delay builds slightly if multiple branches need to be built. Just didn't see that to be particularly useful.
